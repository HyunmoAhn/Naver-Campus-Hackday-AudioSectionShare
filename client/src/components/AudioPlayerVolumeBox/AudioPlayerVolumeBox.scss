.AudioPlayerVolumeBox {
  margin-top: 5px;
}

.AudioPlayerVolumeBox__volume-controller {
  display: inline-block;
  margin: auto;
  width: 50%;
}

.AudioPlayerVolumeBox__mute-btn {
  padding: 3px;
  margin-right: 5px;
  background-color: transparent;
  border: 0;
  color: #BABABA;
  width: 25px;
  float: left;
}

.AudioPlayerVolumeBox__volume {
  color: #B0B0B0;
  margin: 10px;
}