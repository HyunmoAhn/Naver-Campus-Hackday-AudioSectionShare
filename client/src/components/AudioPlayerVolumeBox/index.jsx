import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './AudioPlayerVolumeBox.scss';

const propTypes = {
	volume: PropTypes.number,
	onToggleMute: PropTypes.func,
	onVolumeChange: PropTypes.func,
};

const defaultProps = {
	volume: 0,
	onToggleMute() {},
	onVolumeChange() {},
};

class AudioPlayerVolumeBox extends React.Component {
	render() {
		const { volume, onToggleMute, onVolumeChange} = this.props;
		const volumeClassName = cx('fa', {
			'fa-volume-up': volume > 30,
			'fa-volume-down': volume <= 30 && volume > 0,
			'fa-volume-off': volume <= 0,
		});

		return (
			<div className="AudioPlayerVolumeBox col-xs-4">
				<button
					className="AudioPlayerVolumeBox__mute-btn"
					type="button"
					onClick={onToggleMute}
				>
					<i className={volumeClassName} />
				</button>
				<div
					className="AudioPlayerVolumeBox__volume-controller range__container"
				>
					<input
						className="range"
						type="range"
						value={volume}
						onChange={e => onVolumeChange(e.target.value)}
					/>
				</div>
				<span className="AudioPlayerVolumeBox__volume">{volume}</span>
			</div>
		);
	}
}

AudioPlayerVolumeBox.propTypes = propTypes;
AudioPlayerVolumeBox.defaultProps = defaultProps;

export default AudioPlayerVolumeBox;
