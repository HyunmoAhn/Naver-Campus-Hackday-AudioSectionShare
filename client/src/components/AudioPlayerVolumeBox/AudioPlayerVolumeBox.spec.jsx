import React from 'react';
import { shallow } from 'enzyme';

import AudioPlayerVolumeBox from './';

describe('<AudioPlayerVolumeBox />', () => {
	it('should match to snapshot when render default', () => {
		const wrapper = shallow(<AudioPlayerVolumeBox />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should match volume-icon className when change props.volume', () => {
		const wrapper = shallow(<AudioPlayerVolumeBox />);
		let volumeIcon = wrapper.find('[data-name="volume-icon"]');

		wrapper.setProps({ volume: 0 });
		expect(volumeIcon.props().className).toBe('fa fa-volume-off');

		wrapper.setProps({ volume: 10 });
		volumeIcon = wrapper.find('[data-name="volume-icon"]');
		expect(volumeIcon.props().className).toBe('fa fa-volume-down');

		wrapper.setProps({ volume: 35 });
		volumeIcon = wrapper.find('[data-name="volume-icon"]');
		expect(volumeIcon.props().className).toBe('fa fa-volume-up');
	});

	it('should call onToggleMute when click .AudioPlayerVolumeBox__mute-btn', () => {
		const onToggleMute = jest.fn();
		const wrapper = shallow(<AudioPlayerVolumeBox onToggleMute={onToggleMute} />);
		const muteBtn = wrapper.find('.AudioPlayerVolumeBox__mute-btn');

		muteBtn.simulate('click');
		expect(onToggleMute.mock.calls.length).toBe(1);
	});

	it('should call onVolumeChange when change .range', () => {
		const onVolumeChange = jest.fn();
		const wrapper = shallow(<AudioPlayerVolumeBox onVolumeChange={onVolumeChange} />);
		const range = wrapper.find('.range');

		range.simulate('change', {
			target: {
				value: 50,
			},
		});

		expect(onVolumeChange.mock.calls.length).toBe(1);
		expect(onVolumeChange.mock.calls[0]).toEqual([50]);
	});
});