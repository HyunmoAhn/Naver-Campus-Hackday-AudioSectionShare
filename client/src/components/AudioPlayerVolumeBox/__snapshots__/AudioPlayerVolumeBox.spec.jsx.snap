// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioPlayerVolumeBox /> should match to snapshot when render default 1`] = `
<div
  className="AudioPlayerVolumeBox col-xs-4"
>
  <button
    className="AudioPlayerVolumeBox__mute-btn"
    onClick={[Function]}
    type="button"
  >
    <i
      className="fa fa-volume-off"
      data-name="volume-icon"
    />
  </button>
  <div
    className="AudioPlayerVolumeBox__volume-controller range__container"
  >
    <input
      className="range"
      onChange={[Function]}
      type="range"
      value={0}
    />
  </div>
  <span
    className="AudioPlayerVolumeBox__volume"
  >
    0
  </span>
</div>
`;
